qiskit/chemistry/algorithms/minimum_eigen_solvers/oovqe.py

```diff
@@ -0,0 +1,604 @@
+                                                            qmolecule.mo_coeff)
+        # support for unrestricted spins
+        if qmolecule.mo_coeff_b is not None and qmolecule.x_dip_ints is not None:
+            qmolecule.x_dip_mo_ints_b = qmolecule.oneeints2mo(qmolecule.x_dip_ints,
+                                                              qmolecule.mo_coeff_b)
+            qmolecule.y_dip_mo_ints_b = qmolecule.oneeints2mo(qmolecule.y_dip_ints,
+                                                              qmolecule.mo_coeff_b)
+            qmolecule.z_dip_mo_ints_b = qmolecule.oneeints2mo(qmolecule.z_dip_ints,
+                                                              qmolecule.mo_coeff_b)
+
+
+class OrbitalRotation:
+    r"""
+    Class that regroups methods for creation of matrices that rotate the MOs.
+    It allows to create the unitary matrix U = exp(-kappa) that is parameterized with kappa's
+    elements. The parameters are the off-diagonal elements of the anti-hermitian matrix kappa.
+    """
+
+    def __init__(self,
+                 num_qubits: int,
+                 core: Optional[LegacyBaseOperator] = None,
+                 qmolecule: Optional[QMolecule] = None,
+                 orbital_rotations: list = None,
+                 orbital_rotations_beta: list = None,
+                 parameters: list = None,
+                 parameter_bounds: list = None,
+                 parameter_initial_value: float = 0.1,
+                 parameter_bound_value: tuple = (-2 * np.pi, 2 * np.pi)) -> None:
+
+        """
+        Args:
+            num_qubits: number of qubits necessary to simulate a particular system.
+            core: instance of the :class:`~qiskit.chemistry.core.Hamiltonian` class to
+                 make new qubit operator after the orbital rotation using the data stored in
+                 qmolecule.
+            qmolecule: instance of the :class:`~qiskit.chemistry.QMolecule` class which has methods
+                needed to recompute one-/two-electron/dipole integrals after orbital rotation
+                (C = C0 * exp(-kappa)).
+            orbital_rotations: list of alpha orbitals that are rotated (i.e. [[0,1], ...] the
+                0-th orbital is rotated with 1-st, which corresponds to non-zero entry 01 of
+                the matrix kappa).
+            orbital_rotations_beta: list of beta orbitals that are rotated.
+            parameters: orbital rotation parameter list of matrix elements that rotate the MOs,
+                each associated to a pair of orbitals that are rotated
+                (non-zero elements in matrix kappa), or elements in the orbital_rotation(_beta)
+                lists.
+            parameter_bounds: parameter bounds
+            parameter_initial_value: initial value for all the parameters.
+            parameter_bound_value: value for the bounds on all the parameters
+        """
+
+        self.num_qubits = num_qubits
```

Comment:
Much of these variables should be private I think

Reply:
Yep, done.